nemo/core/neural_factory.py

```diff
@@ -194,50 +384,120 @@ def create_optimizer(self, optimizer, things_to_optimize, optimizer_params):
+    #             else:
+    #                 raise ValueError(
+    #                     "Callback was not a child of ActionCallback nor NeMoCallback and was not understood"
+    #                 )
+
+    # def _perform_on_epoch_end(self, callbacks):
+    #     if callbacks is not None and isinstance(callbacks, List) and len(callbacks) > 0:
+    #         for callback in callbacks:
+    #             if isinstance(callback, ActionCallback):
+    #                 callback.on_epoch_end()
+    #             elif isinstance(callback, NeMoCallback):
+    #                 callback.on_epoch_end(self.state)
+    #             else:
+    #                 raise ValueError(
+    #                     "Callback was not a child of ActionCallback nor NeMoCallback and was not understood"
+    #                 )
+
+    # def _perform_on_batch_start(self, callbacks):
+    #     if callbacks is not None and isinstance(callbacks, List) and len(callbacks) > 0:
+    #         for callback in callbacks:
+    #             if isinstance(callback, ActionCallback):
+    #                 continue
+    #             elif isinstance(callback, NeMoCallback):
+    #                 callback.on_epoch_start(self.state)
+    #             else:
+    #                 raise ValueError(
+    #                     "Callback was not a child of ActionCallback nor NeMoCallback and was not understood"
+    #                 )
+
+    # def _perform_on_batch_end(self, callbacks):
+    #     if callbacks is not None and isinstance(callbacks, List) and len(callbacks) > 0:
+    #         for callback in callbacks:
+    #             if isinstance(callback, ActionCallback):
+    #                 continue
+    #             elif isinstance(callback, NeMoCallback):
+    #                 callback.on_epoch_end(self.state)
+    #             else:
+    #                 raise ValueError(
+    #                     "Callback was not a child of ActionCallback nor NeMoCallback and was not understood"
+    #                 )
+
+    # def _init_callbacks(self, callbacks):
+    #     if callbacks is not None and isinstance(callbacks, List) and len(callbacks) > 0:
+    #         for callback in callbacks:
+    #             if isinstance(callback, ActionCallback):
+    #                 callback.action = self
+
+    # def _update_callbacks(
+    #     self, callbacks=None, registered_tensors=None, final_loss=None,
+    # ):
+    #     # if self.local_rank is None or self.local_rank == 0:
+    #     if callbacks is not None and isinstance(callbacks, List) and len(callbacks) > 0:
+    #         for callback in callbacks:
+    #             if isinstance(callback, ActionCallback):
+    #                 callback._registered_tensors = registered_tensors
+    #             else:  # For now, we can use the old callback function. In the future we should improve this
+    #                 self.training_state.tensor_dict["loss"] = final_loss
```

Comment:
I honestly believe that leaving hundreds of commented code is not a good practice. This is not a production-level code.

Reply:
This has been removed